code/game/gamemodes/dynamic/dynamic.dm

```diff
@@ -680,38 +680,63 @@ GLOBAL_VAR_INIT(dynamic_forced_threat_level, -1)
 		return
 
 	if (forced_latejoin_rule)
-		forced_latejoin_rule.candidates = list(newPlayer)
-		forced_latejoin_rule.trim_candidates()
-		forced_latejoin_rule.load_templates()
-		log_dynamic("Forcing ruleset [forced_latejoin_rule]")
-		if (forced_latejoin_rule.ready(TRUE))
-			if (!forced_latejoin_rule.repeatable)
-				latejoin_rules = remove_from_list(latejoin_rules, forced_latejoin_rule.type)
-			addtimer(CALLBACK(src, TYPE_PROC_REF(/datum/game_mode/dynamic/, execute_midround_latejoin_rule), forced_latejoin_rule), forced_latejoin_rule.delay)
-		forced_latejoin_rule = null
+		log_dynamic("Forcing specific [forced_latejoin_rule.ruletype] ruleset [forced_latejoin_rule].")
+		if(!handle_executing_latejoin(forced_latejoin_rule, newPlayer, forced = TRUE))
+			message_admins("The forced latejoin ruleset [forced_latejoin_rule.name] couldn't be executed \
```

Comment:
Why is this not `log_dynamic`?

Reply:
`forced_latejoin_rule` is only set by admins via the game panel, so I thought they would want to be warned if it fails (rather than logging it)